src/testing/jest/jest-27-and-under/jest-facade.ts

```diff
@@ -0,0 +1,40 @@
+import { JestFacade } from '../jest-facade';
+import { createJestPuppeteerEnvironment as createJestPuppeteerEnvironment27 } from './jest-environment';
+import { jestPreprocessor as jestPreprocessor27 } from './jest-preprocessor';
+import { createTestRunner as createTestRunner27 } from './jest-runner';
+import { runJest as runJest27 } from './jest-runner';
+import { runJestScreenshot as runJestScreenshot27 } from './jest-screenshot';
+import { jestSetupTestFramework as jestSetupTestFramework27 } from './jest-setup-test-framework';
+
+/**
+ * `JestFacade` implementation for communicating between between this directory's version of Jest and Stencil
+ */
+export class Jest27Stencil extends JestFacade {
```

Comment:
j/w, are there any advantages / disadvantages either way between the approaches of having an ABC and implementations that extend it (_a là_ `JEst27Stencil` here) vs an interface that the actual implemetations then `implement`? classes with the `abstract` keyword do actually have a runtime representation and can have concrete methods on them - right now it looks like the `JestFacade` doesn't have any concrete / non-abstract methods on it so would basically be `class JestFacade { }` after compilation.

Did you give any thought to using an interface to constrain the shape of the facade classes (like `Jest27Stencil` here) instead of an ABC? Maybe we will need to have some behavior on the base class in the future

anyway just sort of thinking through this as I read the PR

Reply:
I did give it some thought, and originally had gone the ABC route because I did have a few concrete implementations in the base class here - but those have obviously been removed since then (hence your question). It's relatively trivial to convert between the two. I say let's go with an interface for now unless we find we really need it later down the road. Done in 4327cfcca